joomla/joomla-cms
# [#43528] [5.4] Smart Search: Fix taxonomy filter overwriting when titles are identical - fix issue #43528

Pull Request for Issue #43528 .

### Summary of Changes

- Modified the storage format of taxonomy filters in the processStaticTaxonomy method
- Changed from using taxonomy title as array key to using numeric indexing
- Prevents overwriting of filters when multiple taxonomies share the same title but belong to different categories

### Testing Instructions

1. Install Joomla with sample data
2. Create multiple Smart Search Filters that use taxonomy filters
3. Create taxonomy items with the same title but in different categories
4. Apply these filters to a Smart Search module or menu item
5. Verify that all matching items are displayed correctly, regardless of their category

### Actual result BEFORE applying this Pull Request

When multiple taxonomy items have the same title but belong to different categories, only the last one is kept in the filter array because the title is used as the array key, causing earlier items with the same title to be overwritten.

### Expected result AFTER applying this Pull Request

All taxonomy items are correctly preserved in the filter array, even when they share the same title, because they are stored using numeric indices instead of using the title as the array key.


### Link to documentations
Please select:
